go.mod

```diff
@@ -202,6 +202,8 @@ require (
 )
 
 replace (
+	// This version is a hard-requirement for content negotiation.
+	github.com/prometheus/common => github.com/TheSpiritXIII/prometheus-common v0.45.0-gmp.0
```

Comment:
question: Is this change necessary for the 2.45 fork?

IIUC we're not supporting secrets management with this PR, correct?

Reply:
I updated the comment to be clearer. Let me know if it doesn't make sense!